packages/semcom-upload-component/lib/components/sem-com-register.machine.ts

```diff
@@ -0,0 +1,363 @@
+    || !event.uploadFormContext.author
+    || !event.uploadFormContext.tag
+    || !event.uploadFormContext.shapes
+    || !event.uploadFormContext.version
+    || event.uploadFormContext.latest === undefined
+    || !event.uploadFormContext.checksum) {
+
+    return of(new DataNotSavedEvent('Received invalid component metadata.'));
+
+  }
+
+  // shapes can be multiple comma separated urls, so split them, then create a quad for each.
+  const shapeQuads = event.uploadFormContext.shapes.split(',').map((shape) =>
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#shape'), namedNode(shape)));
+
+  // create quads for all of the component metadata
+  const quads = [
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'), namedNode('http://semcom.digita.ai/voc#component'),),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#label'), literal(event.uploadFormContext.label)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#description'),  literal(event.uploadFormContext.description)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#author'), literal(event.uploadFormContext.author)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#version'), literal(event.uploadFormContext.version)),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#latest'), literal(String(event.uploadFormContext.latest))),
+    quad(namedNode(event.uploadFormContext.uri), namedNode('http://semcom.digita.ai/voc#tag'), literal(event.uploadFormContext.tag)),
+    ...shapeQuads,
+  ];
+
+  const writer = new Writer();
+
+  writer.addQuads(quads);
+
+  // Hash the component's uri to serve as the file name
+  const shaObj = new jsSHA('SHAKE256', 'TEXT');
+  shaObj.update(event.uploadFormContext.uri);
+  const hash = shaObj.getHash('B64', { outputLen: 256 }).replace(/[+=/]/g, '').toLowerCase();
+
+  // fetch the contents of the pod
+  return from(solidFetch(context.url, { headers: { 'Accept': 'text/turtle' } })).pipe(
+    switchMap((response) => zip(of(response), from(response.text()))),
+    // return a set of all filenames within the pod
+    map(([ response, responseText ]) => new Set(
+      response.status === 200
+        ? new Parser({ format: 'Turtle' }).parse(responseText)
+          .filter((quadFromResponse) => quadFromResponse.object.value === 'http://www.w3.org/ns/ldp#Resource')
+          .filter((quadFromResponse) => quadFromResponse.subject.value !== '')
+          .map((quadFromResponse) => quadFromResponse.subject.value)
+        : undefined
+    )),
+    // if the pod is offline, return an empty set
+    catchError(() => of(new Set<string>())),
+    switchMap((components) => zip(of(components), of(writer))),
+    switchMap(([ components, quadWriter ]) => {
```

Comment:
This is not needed: components is passed through the pipe, and writer is accessible from outside of it.